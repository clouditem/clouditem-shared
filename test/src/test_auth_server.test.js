const fetch = require('node-fetch'),
  sut = require('./../../src/test_auth_server'),
  jwt = require('jsonwebtoken'),
  input = { };
let sut_result = null;

beforeAll(() => sut_result = sut(input));
afterAll(() => sut_result.server.close());

test('returns public_key under url', () => fetch(sut_result.url).then(result => result.text())
  .then(text => expect(text).toMatchSnapshot()));

test.each`
rights
${undefined}
${{public_read: false}}
${{public_write: true}}
${{is_creator: true}}
${{user: 'a user'}}

`('sign $rights ', ({rights}) => sut_result.sign({item_id: 'item_id', audience: 'audience', rights}).then(signed_jwt => expect(jwt.decode(signed_jwt)).toMatchSnapshot({iat: expect.anything(), exp: expect.anything()})));

describe('jwt endpoint', () => {

  let jwt_url = null;
  beforeAll(() => jwt_url = `${sut_result.url.split('/').slice(0, -1)
    .concat(['test_item', 'jwt'])
    .join('/')}?audience=test_audience`);

  test('default public_key matches private key', () => fetch(sut_result.url)
    .then(result => result.text())
    .then(public_key => fetch(jwt_url)
      .then(res => res.text())
      .then(signed_jwt => jwt.verify(signed_jwt, public_key, {audience: 'test_audience'}))
      .then(verify_result => expect(verify_result).toMatchSnapshot({iat: expect.anything(), exp: expect.anything()}))));

});
