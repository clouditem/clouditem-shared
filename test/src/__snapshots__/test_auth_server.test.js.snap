// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jwt endpoint default public_key matches private key 1`] = `
Object {
  "aud": "test_audience",
  "exp": Anything,
  "iat": Anything,
  "is_creator": false,
  "iss": "localhost",
  "item_id": "test_item",
  "public_read": true,
  "public_write": false,
  "user": "Dummy",
}
`;

exports[`returns public_key under url 1`] = `
"-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE5hU0g65QNwW4JKUZ4VfpqfeonCFf
ZDPqIL3DNu4Wh3HFgJnJeAgVn1zbLDM74qH5hsLbjLdDxpxuC5GEdpshxw==
-----END PUBLIC KEY-----"
`;

exports[`sign {"is_creator": true}  1`] = `
Object {
  "aud": "audience",
  "exp": Anything,
  "iat": Anything,
  "is_creator": true,
  "iss": "localhost",
  "item_id": "item_id",
  "public_read": true,
  "public_write": false,
  "user": "Dummy",
}
`;

exports[`sign {"public_read": false}  1`] = `
Object {
  "aud": "audience",
  "exp": Anything,
  "iat": Anything,
  "is_creator": false,
  "iss": "localhost",
  "item_id": "item_id",
  "public_read": false,
  "public_write": false,
  "user": "Dummy",
}
`;

exports[`sign {"public_write": true}  1`] = `
Object {
  "aud": "audience",
  "exp": Anything,
  "iat": Anything,
  "is_creator": false,
  "iss": "localhost",
  "item_id": "item_id",
  "public_read": true,
  "public_write": true,
  "user": "Dummy",
}
`;

exports[`sign {"user": "a user"}  1`] = `
Object {
  "aud": "audience",
  "exp": Anything,
  "iat": Anything,
  "is_creator": false,
  "iss": "localhost",
  "item_id": "item_id",
  "public_read": true,
  "public_write": false,
  "user": "a user",
}
`;

exports[`sign undefined  1`] = `
Object {
  "aud": "audience",
  "exp": Anything,
  "iat": Anything,
  "is_creator": false,
  "iss": "localhost",
  "item_id": "item_id",
  "public_read": true,
  "public_write": false,
  "user": "Dummy",
}
`;
